src/test.ts

```diff
@@ -504,55 +520,65 @@ describe('Convert spaces to tabs', () => {
 });
 
 describe('Trailing spaces', () => {
-  it('One trailing space removed', () => {
-    const before = dedent`
-        # H1 
-        line with one trailing spaces 
-      `;
-    const after = dedent`
-        # H1
-        line with one trailing spaces
-      `;
-    expect(rulesDict['trailing-spaces'].apply(before)).toBe(after);
-  });
-  it('Three trailing whitespaces removed', () => {
-    const before = dedent`
-        # H1   
-        line with three trailing spaces   
-      `;
-    const after = dedent`
-        # H1
-        line with three trailing spaces
-      `;
-    expect(rulesDict['trailing-spaces'].apply(before)).toBe(after);
+  ruleTest({
+    testName: 'One trailing space removed',
+    ruleClass: TrailingSpaces,
```

Comment:
I think I am missing how `TrailingSpaces` is converted over to an object rather than a type. Do you mind explaining how this works with the above logic that is as follows:
``` Typescript
function ruleTest<TOptions extends Options>(args: {
    testName: string,
    ruleClass: typeof RuleBase,
    before: string,
    after: string,
     options?: TOptions
  }) {
  it(args.testName, () => {
    const rule = args.ruleClass.rule;
    expect(rule.apply(args.before, args.options)).toBe(args.after);
  });
}
```
How does specifically the following work `args.ruleClass.rule;`? It is my understanding that an instance of the class needs to be made to access this property.

Reply:
Gotcha. I was not aware that that was how we created Singletons in Typescript. That makes more sense now.